add function of collecting customized statistics to files

jiachenwang opened the pull request.

jiachenwang committed:
add function for collecting customized statistics

jiachenwang committed:
Merge remote-tracking branch 'origin/dev' into jiachen-dev

jiachenwang committed:
add statistics timestamp for every statistics entry

jiachenwang committed:
small change to make headless client compatible with Mac

peetonn commented on cpp/client/src/main.cpp:
```diff
@@ -1,159 +1,178 @@
+
+
+    // collect streams statictics
+    boost::asio::io_service staticticsIo;
```
move inside the "class"

peetonn commented on cpp/client/src/main.cpp:
```diff
@@ -1,159 +1,178 @@
+    // collect streams statictics
+    boost::asio::io_service staticticsIo;
+    LogDebug("") << "statisticsSampleInterval(s): " << statisticsSampleInterval << std::endl;
+    collectStreamsStatictics collectStatictics(staticticsIo, headlessAppOnlineTimeSec, statisticsSampleInterval, headlessParams.statistics_, remoteStreamsPrefix, ndnp);
```
1/ let's think of it as a "StatPoller" or "StatCollector"
2/ no need to provide `headlessAppOnlineTimeSec`

peetonn replied:
statCollector.start();

peetonn commented on cpp/client/src/statistics.h:
```diff
@@ -0,0 +1,117 @@
+   std::vector<std::ofstream*> streamStatisticsFilesPointer_;
+};
+
+class streamsStatisticsFiles {
```
use typedef if you want to have nicer name than std::vector

peetonn commented on cpp/client/src/statistics.h:
```diff
@@ -0,0 +1,117 @@
+
+#include "config.h"
+
+class streamStatisticsFiles {
```
why not to use map???

peetonn commented on cpp/client/src/statistics.cpp:
```diff
@@ -0,0 +1,193 @@
+        string remoteStreamPrefix = remoteStreamsPrefix_[remoteStreamsPrefixCount];
+        string remoteStreamPrefixShort = remoteStreamPrefix;
+        streamStatisticsFiles singleStreamStatisticsFiles;
+        StatisticsStorage remoteStreamStatistics = ndnp_->getRemoteStreamStatistics(remoteStreamsPrefix_[remoteStreamsPrefixCount]);
```
no need to get actual statistics (there could be no streams yet...)

peetonn commented on cpp/client/src/statistics.cpp:
```diff
@@ -0,0 +1,193 @@
+        streamStatisticsFiles singleStreamStatisticsFiles;
+        StatisticsStorage remoteStreamStatistics = ndnp_->getRemoteStreamStatistics(remoteStreamsPrefix_[remoteStreamsPrefixCount]);
+
+        makeRemoteStreamPrefixNicer(remoteStreamPrefixShort);
```
maybe `streamPrefixToFileName`? 
string fileName = streamPrefixToFileName(prefix);

peetonn commented on cpp/client/src/statistics.cpp:
```diff
@@ -0,0 +1,193 @@
+            statisticFileName.append(statisticFileNamePrefix);
+
+            statisticFileName.append(remoteStreamPrefixShort);
+            statisticFileName.append(".stat");
```
so maybe `filename = getStatFilename(statFilenamePrefix, streamPrefix);`

peetonn commented on cpp/client/src/main.cpp:
```diff
@@ -1,159 +1,178 @@
+    staticticsIo.stop();
+
+    // sleep(headlessAppOnlineTimeSec);
+    removeRemoteStreams(ndnp, remoteStreamsPrefix);
```
statCollector.stop();

jiachenwang committed:
attampt to publish media streams and save video frames to file

jiachenwang committed:
consumer: video frames write to file(BGRA), producer: read frames from file and publish (bug with frame content, receiving black frames)

jiachenwang committed:
fixed unexpected git conflict

peetonn committed:
'dev' -> 'jiachen-dev'

peetonn committed:
few updates to comply with the latest code from 'dev' branch

peetonn committed:
few formatting updates

peetonn committed:
'dev' -> 'jiachen-dev'

peetonn committed:
removed old gtests, added gtests as submodule, added unit tests for headless app params, repo cleanup

peetonn committed:
headless app parameters code

peetonn committed:
added client class; updated NDN-RTC interface

peetonn committed:
'dev'->'jiachen-dev'

peetonn committed:
implemented statistics collector for consumer

peetonn committed:
resolve ambiguity in NDN-CPP call

peetonn committed:
'dev'->'jiachen-dev'

peetonn committed:
fixed incorrect extraction of stream name from a stream prefix. updated tests

peetonn committed:
added consumer video sink (renderer)

peetonn committed:
removed deprecated "demoapp"

peetonn committed:
added file reader/writer

peetonn committed:
implemented precise signal generator

OSX has imprecise timers. steady_timer on OSX doesn’t work the same way
as it works on Ubuntu. had to implement custom generator which
maintains requested framerate platform-independent.

peetonn committed:
fixed frame file source

reading from frame source does not throw

peetonn committed:
added producer logic on client

peetonn committed:
added producer functionality fixed + its' tests

peetonn committed:
added configs for sample consumer and producer. fix bugs

peetonn committed:
updated unit tests for confg load

peetonn committed:
fixed random deadlocking on background thread. minor fixes in headless app

peetonn committed:
fixed check for re-expressing timed out Key frames

peetonn committed:
added use of refreshNonce for interests
